
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>5.2 Expected IBD of close relatives &#8212; CSE284 - Personal Genomics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!-- 
    this give us a css class that will be invisible only if js is disabled 
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- So that users can add custom icons -->
  <script src="../../_static/scripts/fontawesome.js?digest=26a4bc78f4c0ddb94549"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549" />

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'ancestry/relfind/ibd_pedigree';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="5.3 Computing expected IBD - plink method" href="ibd_plink.html" />
    <link rel="prev" title="5.1 Identity by descent" href="ibd.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="" />
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../_static/logo.png" class="logo__image only-light" alt="CSE284 - Personal Genomics - Home"/>
    <img src="../../_static/logo.png" class="logo__image only-dark pst-js-only" alt="CSE284 - Personal Genomics - Home"/>
  
  
</a></div>
        <div class="sidebar-primary-item">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../index.html">
                    Personal Genomics for Bioinformaticians
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 1 - Introduction</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/intro_to_your_genome.html">Chapter 1: Introduction to the human genome</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/structure_DNA.html">1.1 The structure of DNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/genome_organization.html">1.2 Organization of the human genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/central_dogma.html">1.3 Genes and the central dogma</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/noncoding.html">1.4 The non-coding genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/genetic_variation_terminology.html">1.5 Genetic variation terminology</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/intro_genome_tech.html">Chapter 2: Overview of technologies for genome analysis</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/genotype_arrays.html">2.1 Genotyping arrays</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/ngs.html">2.2 Next-generation sequencing</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/long_reads.html">2.3 Long-read technologies</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/tech_summary.html">2.4 Summary of genomics technologies</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/genome_file_formats.html">Tutorial 1: file formats for describing genetic variation</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/vcf.html">T1.2 VCF files</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/plink.html">T1.2 Plink files</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 2 - Ancestry</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../intro_popgen.html">Chapter 3: Introduction to population genetics</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../popgen/forces_evolution.html">3.1 The four forces of evolution</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/mutations.html">3.2 Types and patterns of mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/drift.html">3.3 Genetic drift</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/selection.html">3.4 Selection</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/hwe.html">3.5 Hardy-Weinberg Equilibrium and random mating</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/fst.html">3.6 Measuring population differentiation with Fst</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/recombination.html">3.7 Recombination</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/WF-simulations.html">Activity: Exploring the Wright-Fisher model of genetic drift</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../global_ancestry.html">Chapter 4: Global ancestry</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../global/pca_global_ancestry.html">4.1 Global ancestry analysis with PCA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../global/admixture_global_ancestry.html">4.2 Global ancestry analysis with ADMIXTURE</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../relative_finding.html">Chapter 5: Relative finding</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="ibd.html">5.1 Identity by descent</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">5.2 Expected IBD of close relatives</a></li>
<li class="toctree-l2"><a class="reference internal" href="ibd_plink.html">5.3 Computing expected IBD - plink method</a></li>
<li class="toctree-l2"><a class="reference internal" href="ibd_segments.html">5.4 Computing expected IBD segment sharing</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../plink_1000G.html">Tutorial 2: Working with 1000 Genomes data and plink</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../plink1kg/1000G.html">T2.1 The 1000 Genomes Project dataset (and wrangling VCF files)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../plink1kg/plink.html">T2.2 Some tips and examples for using plink</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 3 - From genotypes to phenotypes</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../../complextraits/gwas.html">Chapter 7: GWAS for complex traits</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/quantitative_traits.html">7.1 GWAS for quantitative traits</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/GWAS-simulations.html">7.2 Exploring GWAS association statistics</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/confounding.html">7.3 Confounding factors in GWAS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/case_control_traits.html">7.4 GWAS for case-control traits</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/lmms.html">7.5 Linear mixed models</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/power.html">7.6 Power to detect associations</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../complextraits/heritability.html">Chapter 8: Heritability</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/heritability/relatives.html">8.1 Measuring heritability in related individuals</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/heritability/lmms-h2.html">8.2 Measuring SNP-based heritability using LMMs</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../complextraits/prs.html">Chapter 9: Polygenic risk scores</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/prs/clumping-thresholding.html">9.1 Clumping + Threshold (C+T) method</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/prs/bayesian_prs.html">9.2 Bayesian methods for polygenic risk scores</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Appendix 1 - glossary and notation</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../appendix/unix.html">Chapter A1.1: Unix cheat sheet</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/gymrek-lab/personal-genomics-textbook" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/gymrek-lab/personal-genomics-textbook/issues/new?title=Issue%20on%20page%20%2Fancestry/relfind/ibd_pedigree.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/ancestry/relfind/ibd_pedigree.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button>


<button class="btn btn-sm pst-navbar-icon search-button search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
</button>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>5.2 Expected IBD of close relatives</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#summary-of-ibd-sharing">Summary of IBD sharing</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#unrelateds-d-and-e">Unrelateds (D and E)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#siblings-a-and-b">Siblings (A and B)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#parent-child-f-and-b">Parent-child (F and B)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#grandparent-grandchild-f-and-c">Grandparent/grandchild (F and C)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#avuncular-a-and-c">Avuncular (A and C)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#greatgrandparent-greatgrandchild-f-and-e">Greatgrandparent/greatgrandchild (F and E)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#first-cousins-g-and-c-and-beyond">First cousins (G and C) (and beyond)</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="expected-ibd-of-close-relatives">
<h1>5.2 Expected IBD of close relatives<a class="headerlink" href="#expected-ibd-of-close-relatives" title="Link to this heading">#</a></h1>
<p>Different relationships will show characteristic percentages of their genomes shared at IBD=0, 1, or 2. The table below summarizes some of those relationships. In the sections below, we work out how you could derive some of these.</p>
<section id="summary-of-ibd-sharing">
<h2>Summary of IBD sharing<a class="headerlink" href="#summary-of-ibd-sharing" title="Link to this heading">#</a></h2>
<div class="pst-scrollable-table-container"><table class="table">
<thead>
<tr class="row-odd"><th class="head"><p>Relationship</p></th>
<th class="head"><p>%IBD=2</p></th>
<th class="head"><p>%IBD=1</p></th>
<th class="head"><p>%IBD=0</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Identical twins</p></td>
<td><p>1</p></td>
<td><p>0</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><p>Full siblings</p></td>
<td><p>0.25</p></td>
<td><p>0.50</p></td>
<td><p>0.25</p></td>
</tr>
<tr class="row-even"><td><p>Parent-child</p></td>
<td><p>0</p></td>
<td><p>1</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><p>Half siblings</p></td>
<td><p>0</p></td>
<td><p>0.5</p></td>
<td><p>0.5</p></td>
</tr>
<tr class="row-even"><td><p>Grandparent/grandchild</p></td>
<td><p>0</p></td>
<td><p>0.5</p></td>
<td><p>0.5</p></td>
</tr>
<tr class="row-odd"><td><p>Avuncular</p></td>
<td><p>0</p></td>
<td><p>0.5</p></td>
<td><p>0.5</p></td>
</tr>
<tr class="row-even"><td><p>First cousins</p></td>
<td><p>0</p></td>
<td><p>0.25</p></td>
<td><p>0.75</p></td>
</tr>
<tr class="row-odd"><td><p>Unrelated</p></td>
<td><p>0</p></td>
<td><p>0</p></td>
<td><p>1</p></td>
</tr>
</tbody>
</table>
</div>
<p>Consider the following pedigree:</p>
<p><img alt="pedigree3" src="../../_images/pedigree_example3.png" /></p>
<p>Let <span class="math notranslate nohighlight">\(m_1\)</span>, <span class="math notranslate nohighlight">\(m_2\)</span>, <span class="math notranslate nohighlight">\(f_1\)</span>, and <span class="math notranslate nohighlight">\(f_2\)</span> be the names of the chromosomes of the grandparents of the pedigree. Below we show how to estimate the percent of the genome that will be shared for different relative pairs.</p>
<p>Note, when doing so, we will compute the probability that any particular region of the genome will be shared at IBD=0, 1, or 2. This will be equal to the total percent of the genome we expect to be shared at these IBD levels, since we can think about each region of the genome independently. Therefore, we will be able to ignore recombination when computing these percentages.</p>
</section>
<section id="unrelateds-d-and-e">
<h2>Unrelateds (D and E)<a class="headerlink" href="#unrelateds-d-and-e" title="Link to this heading">#</a></h2>
<p>Generally we can assume unrelated individuals (such as D and E above) do not share any genomic segments and share 100% of their genomes at IBD=0. (Note, in the case of populations with high rates of inbreeding or very low effective population size this will not be the case. Further, in most populations there will be some degree of background genome sharing even for “unrelated” individuals, contributed by very distantly related ancestors).</p>
</section>
<section id="siblings-a-and-b">
<h2>Siblings (A and B)<a class="headerlink" href="#siblings-a-and-b" title="Link to this heading">#</a></h2>
<p>Let:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(S\)</span> be the set of possible child chromosomes (<span class="math notranslate nohighlight">\(S \in \{(m_1, f_1), (m_1, f_2), (m_2, f_1), (m_2, f_2)\})\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(IBD(a,b)\)</span> be the IBD sharing of chromosome combinations <span class="math notranslate nohighlight">\(a\)</span> and <span class="math notranslate nohighlight">\(b\)</span>. e.g.:</p>
<ul>
<li><p><span class="math notranslate nohighlight">\(IBD((m_1, f_1), (m_1, f_2)) = 1\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(IBD((m_2, f_1), (m_1, f_2)) = 0\)</span></p></li>
<li><p>etc.</p></li>
</ul>
</li>
</ul>
<p>Then we can compute the probability that <span class="math notranslate nohighlight">\(A\)</span> and <span class="math notranslate nohighlight">\(B\)</span> share a segment at <span class="math notranslate nohighlight">\(IBD=i\)</span>:</p>
<div class="math notranslate nohighlight">
\[P(IBD[A,B]=i) = \sum_{a \in S, b \in S s.t. IBD(a,b)=i} P(A=a)P(B=b)\]</div>
<p>Any of the four combinations in <span class="math notranslate nohighlight">\(S\)</span> are equally likely in each sibling, so <span class="math notranslate nohighlight">\(P(A=a)\)</span> and <span class="math notranslate nohighlight">\(P(B=b)\)</span> will both be 0.25 in all cases.</p>
<p>For <span class="math notranslate nohighlight">\(P(IBD[A,B]=0)\)</span>, the following combinations have IBD=0:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\((m_1, f_1), (m_2, f_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\((m_1, f_2), (m_2, f_1)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\((m_2, f_1), (m_1, f_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\((m_2, f_2), (m_1, f_1)\)</span></p></li>
</ul>
<p>So we get <span class="math notranslate nohighlight">\(P(IBD[A,B]=0) = 4*0.25*0.25 = 0.25\)</span>.</p>
<p>Similarly for <span class="math notranslate nohighlight">\(P(IBD[A,B]=2)\)</span> you can come up with four different combinations that give IBD=2:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\((m_1, f_1), (m_1, f_1)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\((m_1, f_2), (m_1, f_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\((m_2, f_1), (m_2, f_1)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\((m_2, f_2), (m_2, f_2)\)</span></p></li>
</ul>
<p>and similarly get <span class="math notranslate nohighlight">\(P(IBD[A,B]=2) = 4*0.25*0.25 = 0.25\)</span>.</p>
<p>You can follow the same procedure (or just subtract <span class="math notranslate nohighlight">\(1-P(IBD=0)-P(IBD=2))\)</span> to come up with <span class="math notranslate nohighlight">\(P(IBD[A,B]=1)=0.5\)</span>.</p>
<p><strong>To summarize, for siblings with the %IBD=0, 1, 2 is 0.25, 0.5, 0.25.</strong></p>
</section>
<section id="parent-child-f-and-b">
<h2>Parent-child (F and B)<a class="headerlink" href="#parent-child-f-and-b" title="Link to this heading">#</a></h2>
<p>Similar to in the above let:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(S\)</span> be the set of possible child chromosomes (<span class="math notranslate nohighlight">\(S \in \{(m_1, f_1), (m_1, f_2), (m_2, f_1), (m_2, f_2)\})\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(F\)</span> be the set of possible father (paternal) chromosome combinations (<span class="math notranslate nohighlight">\(F \in \{(f_1, f_2)\}\)</span>) (there is only one possibility here)</p></li>
<li><p><span class="math notranslate nohighlight">\(IBD(a,b)\)</span> is defined the same as above.</p></li>
</ul>
<p>Then:</p>
<div class="math notranslate nohighlight">
\[P(IBD[F,B]=i) = \sum_{f \in F, b \in S s.t. IBD(f,b)=i} P(F=f)P(B=b)\]</div>
<p>Since there is only one possible father combination ((<span class="math notranslate nohighlight">\(f_1,f_2\)</span>) with probability 1), <span class="math notranslate nohighlight">\(P(F=f)\)</span> is always 1 here. As in the above case, any of the four combinations in <span class="math notranslate nohighlight">\(S\)</span> are equally likely, so <span class="math notranslate nohighlight">\(P(B=b)\)</span> will be 0.25 in all cases. We can simplify to:</p>
<div class="math notranslate nohighlight">
\[P(IBD[F,B]=i) = \sum_{f \in F, b \in S s.t. IBD(f, b)=i} 0.25\]</div>
<p>Further, all four possible child chromosome combinations in <span class="math notranslate nohighlight">\(S\)</span> have IBD of 1 with the father. So we have:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(P(IBD[F,B]=1) = 4(0.25) = 1\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(P(IBD[F,B]=0) = 0\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(P(IBD[F,B]=2) = 0\)</span></p></li>
</ul>
</section>
<section id="grandparent-grandchild-f-and-c">
<h2>Grandparent/grandchild (F and C)<a class="headerlink" href="#grandparent-grandchild-f-and-c" title="Link to this heading">#</a></h2>
<p>We can use a similar concept as above:</p>
<div class="math notranslate nohighlight">
\[P(IBD[F,C]=i) = \sum_{f \in F, c\in C s.t. IBD(f, c)=i}P(F=f)P(C=c)\]</div>
<ul class="simple">
<li><p>The grandparent (<span class="math notranslate nohighlight">\(F\)</span>) has only one possible combination (<span class="math notranslate nohighlight">\(f_1,f_2\)</span>) with probability 1.</p></li>
<li><p>Let <span class="math notranslate nohighlight">\(U=(u_1,u_2)\)</span> denote the chromosomes of the unrelated parent of the grandchild (the one not descended from F).</p></li>
<li><p>The grandchild (<span class="math notranslate nohighlight">\(C\)</span>) will have the following possible combinations with equal probability (you can work this out if you assume the father, <span class="math notranslate nohighlight">\(B\)</span>, has each possible combination in <span class="math notranslate nohighlight">\(S\)</span> with equal probability):</p></li>
</ul>
<div class="math notranslate nohighlight">
\[C \in \{(m_1, u_1), (m_1, u_2), (m_2, u_2), (m_2, u_2)\]</div>
<div class="math notranslate nohighlight">
\[... (f_1, u_1), (f_1, u_2), (f_2, u_1), (f_2, u_2)\}\]</div>
<p>So the above simplifies to:</p>
<div class="math notranslate nohighlight">
\[P(IBD[F,C]=i) = \sum_{f \in F, c\in C s.t. IBD(f,c)=i}0.125\]</div>
<ul class="simple">
<li><p>There are no combinations in <span class="math notranslate nohighlight">\(C\)</span> that give IBD=2 with <span class="math notranslate nohighlight">\(F\)</span>, so <span class="math notranslate nohighlight">\(P(IBD[F,C]=2)=0\)</span></p></li>
<li><p>There are 4 combinations that give IBD=1, so <span class="math notranslate nohighlight">\(P(IBD[F,C]=1) = 4(0.125) = 0.5\)</span></p></li>
<li><p>There are 4 combinations that give IBD=0, so <span class="math notranslate nohighlight">\(P(IBD[F,C]=0) = 4(0.125) = 0.5\)</span></p></li>
</ul>
</section>
<section id="avuncular-a-and-c">
<h2>Avuncular (A and C)<a class="headerlink" href="#avuncular-a-and-c" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(A\)</span> will have any of the following chromosome combinations with equal probability: (<span class="math notranslate nohighlight">\(S \in \{(m_1, f_1), (m_1, f_2), (m_2, f_1), (m_2, f_2)\}\)</span>)</p></li>
<li><p>See above for the 8 possible combinations <span class="math notranslate nohighlight">\(C\)</span> could have.</p></li>
</ul>
<p>So our equation above will simplify to:</p>
<div class="math notranslate nohighlight">
\[P(IBD[A,C]=i) = \sum_{a \in S, c \in C s.t. IBD(a,c)=i}0.25(0.125)\]</div>
<p>We can tell right away none of the possible combinations have IBD=2.</p>
<p>There are 16 combinations that will give IBD=1:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(A=(m_1,f_1), C=(m_1,u_1)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_1,f_1), C=(m_1,u_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_1,f_1), C=(f_1,u_1)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_1,f_1), C=(f_1,u_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_2,f_1), C=(m_2,u_1)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_2,f_1), C=(m_2,u_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_2,f_1), C=(f_1,u_1)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_2,f_1), C=(f_1,u_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_1,f_2), C=(m_1,u_1)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_1,f_2), C=(m_1,u_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_1,f_2), C=(f_2,u_1)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_1,f_2), C=(f_2,u_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_2,f_2), C=(m_2,u_1)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_2,f_2), C=(m_2,u_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_2,f_2), C=(f_2,u_1)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(A=(m_2,f_2), C=(f_2,u_2)\)</span></p></li>
</ul>
<p>so we have <span class="math notranslate nohighlight">\(P(IBD[A,C]=1)=16(0.25)=0.5\)</span>. Similarly you can get <span class="math notranslate nohighlight">\(P(IBD[A,C]=0)=0.5\)</span>.</p>
</section>
<section id="greatgrandparent-greatgrandchild-f-and-e">
<h2>Greatgrandparent/greatgrandchild (F and E)<a class="headerlink" href="#greatgrandparent-greatgrandchild-f-and-e" title="Link to this heading">#</a></h2>
<p>Eventually, it becomes cumbersome to write down all those combinations, but we could use the same method as above. Instead we will describe the following examples using intuition.</p>
<p>First, we know IBD=2 will always have 0 probability in this case, since E has one parent unrelated to F. So we really have to just decide about IBD1 vs. IBD0.</p>
<p>We know that the child (B) will inherit one chromosome from F. The grandchild (C) will inherit that chromosome with 50% probability. If C gets a chromosome from F, they will pass it on to the greatgrandchild (E) with 50% probability.</p>
<p>So P(IBD=1) is 0.5(0.5) = 0.25 and P(IBD=0) = 1-0.25 = 0.75</p>
<p>We could extend this to great-great-grandchild to get P(IBD=1) = <span class="math notranslate nohighlight">\(0.5*0.5*0.5\)</span> = 0.125 and so on.</p>
</section>
<section id="first-cousins-g-and-c-and-beyond">
<h2>First cousins (G and C) (and beyond)<a class="headerlink" href="#first-cousins-g-and-c-and-beyond" title="Link to this heading">#</a></h2>
<p>Again we will use intuition for this one. Each grandchild will inherit one of the four grandparent chromosomes. There is a 25% chance it will be the same one. So P(IBD=1)=0.25.</p>
<p>If we wanted to do G and E:
G will inherit one of the four grandparent chromosomes
C will inherit one of the four grandparent chromosomes with 50% chance.
There is a 25% chance it will be the same one</p>
<p>So: P(IBD=1) = 0.5(0.25)=0.125</p>
<p>Each subsequent generation, each copy of the grandparent chromosomes will be lost with 50% probability. So we can get the probability that <span class="math notranslate nohighlight">\(n\)</span>th cousins share a segment IBD=1 is <span class="math notranslate nohighlight">\(\frac{1}{2^{2n}}\)</span>.</p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./ancestry/relfind"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="ibd.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">5.1 Identity by descent</p>
      </div>
    </a>
    <a class="right-next"
       href="ibd_plink.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">5.3 Computing expected IBD - plink method</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#summary-of-ibd-sharing">Summary of IBD sharing</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#unrelateds-d-and-e">Unrelateds (D and E)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#siblings-a-and-b">Siblings (A and B)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#parent-child-f-and-b">Parent-child (F and B)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#grandparent-grandchild-f-and-c">Grandparent/grandchild (F and C)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#avuncular-a-and-c">Avuncular (A and C)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#greatgrandparent-greatgrandchild-f-and-e">Greatgrandparent/greatgrandchild (F and E)</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#first-cousins-g-and-c-and-beyond">First cousins (G and C) (and beyond)</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Melissa Gymrek
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549"></script>
<script defer src="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>