
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>T2.1 The 1000 Genomes Project dataset (and wrangling VCF files) &#8212; CSE284 - Personal Genomics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!-- 
    this give us a css class that will be invisible only if js is disabled 
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- So that users can add custom icons -->
  <script src="../../_static/scripts/fontawesome.js?digest=26a4bc78f4c0ddb94549"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549" />

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'ancestry/plink1kg/1000G';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="T2.2 Some tips and examples for using plink" href="plink.html" />
    <link rel="prev" title="Tutorial 2: Working with 1000 Genomes data and plink" href="../plink_1000G.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="" />
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../_static/logo.png" class="logo__image only-light" alt="CSE284 - Personal Genomics - Home"/>
    <img src="../../_static/logo.png" class="logo__image only-dark pst-js-only" alt="CSE284 - Personal Genomics - Home"/>
  
  
</a></div>
        <div class="sidebar-primary-item">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../index.html">
                    Personal Genomics for Bioinformaticians
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 1 - Introduction</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/intro_to_your_genome.html">Chapter 1: Introduction to the human genome</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/structure_DNA.html">1.1 The structure of DNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/genome_organization.html">1.2 Organization of the human genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/central_dogma.html">1.3 Genes and the central dogma</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/noncoding.html">1.4 The non-coding genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/genetic_variation_terminology.html">1.5 Genetic variation terminology</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/intro_genome_tech.html">Chapter 2: Overview of technologies for genome analysis</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/genotype_arrays.html">2.1 Genotyping arrays</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/ngs.html">2.2 Next-generation sequencing</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/long_reads.html">2.3 Long-read technologies</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/tech_summary.html">2.4 Summary of genomics technologies</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/genome_file_formats.html">Tutorial 1: file formats for describing genetic variation</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/vcf.html">T1.2 VCF files</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/plink.html">T1.2 Plink files</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 2 - Ancestry</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../intro_popgen.html">Chapter 3: Introduction to population genetics</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../popgen/forces_evolution.html">3.1 The four forces of evolution</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/mutations.html">3.2 Types and patterns of mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/drift.html">3.3 Genetic drift</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/selection.html">3.4 Selection</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/hwe.html">3.5 Hardy-Weinberg Equilibrium and random mating</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/fst.html">3.6 Measuring population differentiation with Fst</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/recombination.html">3.7 Recombination</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/WF-simulations.html">Activity: Exploring the Wright-Fisher model of genetic drift</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../global_ancestry.html">Chapter 4: Global ancestry</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../global/pca_global_ancestry.html">4.1 Global ancestry analysis with PCA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../global/admixture_global_ancestry.html">4.2 Global ancestry analysis with ADMIXTURE</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../relative_finding.html">Chapter 5: Relative finding</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd.html">5.1 Identity by descent</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd_pedigree.html">5.2 Expected IBD of close relatives</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd_plink.html">5.3 Computing expected IBD - plink method</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd_segments.html">5.4 Computing expected IBD segment sharing</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../plink_1000G.html">Tutorial 2: Working with 1000 Genomes data and plink</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2 current active"><a class="current reference internal" href="#">T2.1 The 1000 Genomes Project dataset (and wrangling VCF files)</a></li>
<li class="toctree-l2"><a class="reference internal" href="plink.html">T2.2 Some tips and examples for using plink</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 3 - From genotypes to phenotypes</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../../complextraits/gwas.html">Chapter 7: GWAS for complex traits</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/quantitative_traits.html">7.1 GWAS for quantitative traits</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/GWAS-simulations.html">7.2 Exploring GWAS association statistics</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/confounding.html">7.3 Confounding factors in GWAS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/case_control_traits.html">7.4 GWAS for case-control traits</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/lmms.html">7.5 Linear mixed models</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/power.html">7.6 Power to detect associations</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../complextraits/heritability.html">Chapter 8: Heritability</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/heritability/relatives.html">8.1 Measuring heritability in related individuals</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/heritability/lmms-h2.html">8.2 Measuring SNP-based heritability using LMMs</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../complextraits/prs.html">Chapter 9: Polygenic risk scores</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/prs/clumping-thresholding.html">9.1 Clumping + Threshold (C+T) method</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/prs/bayesian_prs.html">9.2 Bayesian methods for polygenic risk scores</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Appendix 1 - glossary and notation</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../appendix/unix.html">Chapter A1.1: Unix cheat sheet</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/gymrek-lab/personal-genomics-textbook" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/gymrek-lab/personal-genomics-textbook/issues/new?title=Issue%20on%20page%20%2Fancestry/plink1kg/1000G.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/ancestry/plink1kg/1000G.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button>


<button class="btn btn-sm pst-navbar-icon search-button search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
</button>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>T2.1 The 1000 Genomes Project dataset (and wrangling VCF files)</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#overview">Overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#genomes-population-codes">1000 Genomes population codes</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#genomes-data-vcfs">1000 Genomes data - VCFs</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#genomes-data-populations">1000 Genomes data - Populations</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#using-1000-genomes-data-in-plink">Using 1000 Genomes data in plink</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#wrangling-1000-genomes-vcf-files-with-bcftools">Wrangling 1000 Genomes VCF files with bcftools</a></li>
</ul>
</li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="t2-1-the-1000-genomes-project-dataset-and-wrangling-vcf-files">
<h1>T2.1 The 1000 Genomes Project dataset (and wrangling VCF files)<a class="headerlink" href="#t2-1-the-1000-genomes-project-dataset-and-wrangling-vcf-files" title="Link to this heading">#</a></h1>
<section id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Link to this heading">#</a></h2>
<p>The 1000 Genomes Project dataset provides a deep catalog of human genetic variation across healthy individuals from diverse populations.
Unlike nearly all other large human genetic variation datasets, the 1000 Genomes Project data is made available as a community resource. Access is available freely on the internet to any researcher who wants to use the data.</p>
<p>Our analyses throughout the course will focus on the “phase 3” release of the data, which contains genetic variation (VCF files) for 2,504 samples. See <a class="reference external" href="https://www.nature.com/articles/nature15393">A global reference for human genetic variation</a> for more details on how this dataset was generated and for a description of the main findings.</p>
<p>Note, this data is using the GRCh37 human reference genome build. Newer callsets have been generated using the GRCh38 human reference genome, but we will stick with the original phase 3 data here.</p>
<section id="genomes-population-codes">
<h3>1000 Genomes population codes<a class="headerlink" href="#genomes-population-codes" title="Link to this heading">#</a></h3>
<p>Samples originate from distinct population groups.
Populations are broadly grouped by <em>superpopulation</em>, denoted by widely used 3-letter codes (EUR=Europe, SAS=South Asia, EAS=East Asia, AFR=African, AMR=American).
Superpopulations can be further divided into <em>populations</em> with their own 3 letter codes.
For example “YRI” denotes the Yoruban population from Nigeria, which falls under the “AFR” superpopulation.
See a full list of population and superpopulation codes below:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">EAS</span>
<span class="o">---</span>
<span class="n">CHB</span>	<span class="n">Han</span> <span class="n">Chinese</span>
<span class="n">JPT</span>	<span class="n">Japanese</span>
<span class="n">CHS</span>	<span class="n">Southern</span> <span class="n">Han</span> <span class="n">Chinese</span>
<span class="n">CDX</span>	<span class="n">Dai</span> <span class="n">Chinese</span>
<span class="n">KHV</span>	<span class="n">Kinh</span> <span class="n">Vietnamese</span>
<span class="n">CHD</span>	<span class="n">Denver</span> <span class="n">Chinese</span>

<span class="n">EUR</span>
<span class="o">---</span>
<span class="n">CEU</span>	<span class="n">CEPH</span> <span class="p">(</span><span class="n">Utah</span><span class="p">,</span> <span class="n">NW</span> <span class="n">Europe</span><span class="p">)</span>
<span class="n">TSI</span>	<span class="n">Tuscan</span>
<span class="n">GBR</span>	<span class="n">British</span>
<span class="n">FIN</span>	<span class="n">Finnish</span>
<span class="n">IBS</span>	<span class="n">Spanish</span>

<span class="n">AFR</span>
<span class="o">---</span>
<span class="n">YRI</span>	<span class="n">Yoruba</span>
<span class="n">LWK</span>	<span class="n">Luhya</span>
<span class="n">GWD</span>	<span class="n">Gambian</span>
<span class="n">MSL</span>	<span class="n">Mende</span>
<span class="n">ESN</span>	<span class="n">Esan</span>
<span class="n">ASW</span>	<span class="n">African</span><span class="o">-</span><span class="n">American</span> <span class="n">SW</span>
<span class="n">ACB</span>	<span class="n">African</span><span class="o">-</span><span class="n">Caribbean</span>

<span class="n">AMR</span>
<span class="o">---</span>
<span class="n">MXL</span>	<span class="n">Mexican</span><span class="o">-</span><span class="n">American</span>
<span class="n">PUR</span>	<span class="n">Puerto</span> <span class="n">Rican</span>
<span class="n">CLM</span>	<span class="n">Colombian</span>
<span class="n">PEL</span>	<span class="n">Peruvian</span>

<span class="n">SAS</span>
<span class="o">---</span>
<span class="n">GIH</span>	<span class="n">Gujarati</span>
<span class="n">PJL</span>	<span class="n">Punjabi</span>
<span class="n">BEB</span>	<span class="n">Bengali</span> 
<span class="n">STU</span>	<span class="n">Sri</span> <span class="n">Lankan</span>
<span class="n">ITU</span>	<span class="n">Indian</span> <span class="n">Telugu</span>
</pre></div>
</div>
</section>
<section id="genomes-data-vcfs">
<h3>1000 Genomes data - VCFs<a class="headerlink" href="#genomes-data-vcfs" title="Link to this heading">#</a></h3>
<p>We will work with the phase 3 VCF files. These were obtained from <a class="reference external" href="https://www.internationalgenome.org/data">https://www.internationalgenome.org/data</a>. If you are working on the CSE284 datahub, you can find the VCF files (one per chromosome) here:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">~/</span><span class="n">public</span><span class="o">/</span><span class="mi">1000</span><span class="n">Genomes</span><span class="o">/</span>
</pre></div>
</div>
<p>These were downloaded from the <a class="reference external" href="https://hgdownload.cse.ucsc.edu/gbdb/hg19/1000Genomes/phase3/">UCSC Genome Browser</a>.</p>
<p>For each chromosome, you’ll see two files, e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ALL</span><span class="o">.</span><span class="n">chr2</span><span class="o">.</span><span class="n">phase3_shapeit2_mvncall_integrated_v5a</span><span class="mf">.20130502</span><span class="o">.</span><span class="n">genotypes</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span>
<span class="n">ALL</span><span class="o">.</span><span class="n">chr2</span><span class="o">.</span><span class="n">phase3_shapeit2_mvncall_integrated_v5a</span><span class="mf">.20130502</span><span class="o">.</span><span class="n">genotypes</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span><span class="o">.</span><span class="n">tbi</span>
</pre></div>
</div>
<p>The first is a bgzipped VCF file. The second is an index file, created by the <code class="docutils literal notranslate"><span class="pre">tabix</span></code> program. Below are some tips on looking at these files:</p>
<ul class="simple">
<li><p>You can visualize a zipped file on the command line using <code class="docutils literal notranslate"><span class="pre">zcat</span></code> (rather than <code class="docutils literal notranslate"><span class="pre">cat</span></code>). To scroll through the file, e.g.:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">zcat</span> <span class="n">ALL</span><span class="o">.</span><span class="n">chr15</span><span class="o">.</span><span class="n">phase3_shapeit2_mvncall_integrated_v5a</span><span class="mf">.20130502</span><span class="o">.</span><span class="n">genotypes</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="o">|</span> <span class="n">less</span> <span class="o">-</span><span class="n">S</span>
</pre></div>
</div>
<ul class="simple">
<li><p>Once a VCF file is indexed with <code class="docutils literal notranslate"><span class="pre">tabix</span></code>, we can quickly extract a particular region of interest. e.g.:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">tabix</span> <span class="n">ALL</span><span class="o">.</span><span class="n">chr15</span><span class="o">.</span><span class="n">phase3_shapeit2_mvncall_integrated_v5a</span><span class="mf">.20130502</span><span class="o">.</span><span class="n">genotypes</span><span class="o">.</span><span class="n">vcf</span><span class="o">.</span><span class="n">gz</span> <span class="mi">15</span><span class="p">:</span><span class="mi">28341863</span><span class="o">-</span><span class="mi">28341863</span> <span class="o">|</span> <span class="n">less</span> <span class="o">-</span><span class="n">S</span>
</pre></div>
</div>
<p>Note, you’ll see some 1000 Genomes specific fields in the INFO part of the VCF file. For example, you can see the alternate allele frequency of each variant in the different superpopulations in the fields <code class="docutils literal notranslate"><span class="pre">AFR_AF</span></code>, <code class="docutils literal notranslate"><span class="pre">EUR_AF</span></code>, etc.</p>
</section>
<section id="genomes-data-populations">
<h3>1000 Genomes data - Populations<a class="headerlink" href="#genomes-data-populations" title="Link to this heading">#</a></h3>
<p><a class="reference external" href="https://raw.githubusercontent.com/gymrek-lab/personal-genomics-textbook/main/data/igsr_samples.tsv">This file</a> contains information about the population label of each sample. It was downloaded directly from the <a class="reference external" href="https://www.internationalgenome.org/data-portal/sample">IGSR website</a>. If you are on the CSE284 datahub, you can find this file at:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">~/</span><span class="n">public</span><span class="o">/</span><span class="mi">1000</span><span class="n">Genomes</span><span class="o">/</span><span class="n">igsr_samples</span><span class="o">.</span><span class="n">tsv</span>
</pre></div>
</div>
<p>Below are some helpful example commands for extracting population sets from this file:</p>
<ul class="simple">
<li><p>Print a list of samples coming from the CEU population:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cat</span> <span class="n">igsr_samples</span><span class="o">.</span><span class="n">tsv</span> <span class="o">|</span> \
  <span class="n">grep</span> <span class="s2">&quot;1000 Genomes phase 3 release&quot;</span> <span class="o">|</span> \
  <span class="n">awk</span> <span class="o">-</span><span class="sa">F</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span> <span class="s1">&#39;($4==&quot;CEU&quot;)&#39;</span> <span class="o">|</span> \
  <span class="n">awk</span> <span class="s1">&#39;{print $1}&#39;</span>
</pre></div>
</div>
<ul class="simple">
<li><p>Print a list of samples coming from the AFR superpopulation:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cat</span> <span class="n">igsr_samples</span><span class="o">.</span><span class="n">tsv</span> <span class="o">|</span> \
  <span class="n">grep</span> <span class="s2">&quot;1000 Genomes phase 3 release&quot;</span> <span class="o">|</span> \
  <span class="n">awk</span> <span class="o">-</span><span class="sa">F</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span> <span class="s1">&#39;($6==&quot;AFR&quot;)&#39;</span> <span class="o">|</span> \
  <span class="n">awk</span> <span class="s1">&#39;{print $1}&#39;</span>
</pre></div>
</div>
<ul class="simple">
<li><p>Print a list of samples coming from the AFR superpopulation, in the format required for plink (see below):</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cat</span> <span class="n">igsr_samples</span><span class="o">.</span><span class="n">tsv</span> <span class="o">|</span> \
  <span class="n">grep</span> <span class="s2">&quot;1000 Genomes phase 3 release&quot;</span> <span class="o">|</span> \
  <span class="n">awk</span> <span class="o">-</span><span class="sa">F</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span> <span class="s1">&#39;($6==&quot;AFR&quot;)&#39;</span> <span class="o">|</span> \
  <span class="n">awk</span> <span class="s1">&#39;{print $1 &quot;</span><span class="se">\t</span><span class="s1">&quot; $1}&#39;</span>
</pre></div>
</div>
</section>
<section id="using-1000-genomes-data-in-plink">
<h3>Using 1000 Genomes data in plink<a class="headerlink" href="#using-1000-genomes-data-in-plink" title="Link to this heading">#</a></h3>
<p><a class="reference external" href="https://www.cog-genomics.org/plink/1.9/">Plink</a> is a handy tool for doing all sorts of things with genetic data. The newer version of plink (v1.9+) supports operations directly on VCF files, as well as conversion of VCF files to other data formats supported by plink.
Below shows examples of some plink operations on the 1000 Genomes VCF files.</p>
<ul class="simple">
<li><p>Extract data from chr15 for the CEU population</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span># First, make a sample file that can be read by plink
# Historically, plink data includes family IDs and sample IDs for each sample
# We do not have family information, so will use the sample ID for both fields
# See https://www.cog-genomics.org/plink/1.9/filter#indiv
cat igsr_samples.tsv | \
  grep &quot;1000 Genomes phase 3 release&quot; | \
  awk -F&quot;\t&quot; &#39;($4==&quot;CEU&quot;)&#39; | \
  awk &#39;{print $1 &quot;\t&quot; $1}&#39; &gt; ceu_samples.txt

# Now, run plink to subset
# Option info:
# --vcf uses a VCF file as input
# --out gives the prefix of output files
# --make-bed specifies to output plink binary format (BED/BIM/FAM)
# --keep specifies to keep only the samples in the provided file
# --double-id means to use the sample ID from the VCF as the family ID 
# See https://www.cog-genomics.org/plink/1.9/input#vcf for more info
VCF=~/public/1000Genomes/ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz
plink --vcf $VCF \
	--out ceu_chr15 \
	--make-bed \
	--keep ceu_samples.txt \
	--double-id
</pre></div>
</div>
<ul class="simple">
<li><p>Compute Hardy-Weinberg Equilibrium statistics for CEU samples SNP rs12913832</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span># See output file ceu_hardy.hwe
VCF=~/public/1000Genomes/ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz
plink --vcf $VCF \
   --hardy \
   --keep ceu_samples.txt \
   --double-id \
   --snp rs12913832 \
   --out ceu_hardy
</pre></div>
</div>
</section>
<section id="wrangling-1000-genomes-vcf-files-with-bcftools">
<h3>Wrangling 1000 Genomes VCF files with bcftools<a class="headerlink" href="#wrangling-1000-genomes-vcf-files-with-bcftools" title="Link to this heading">#</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">bcftools</span></code> is another package with very useful VCF wrangling capabilities.
You are encouraged to check these out here: <a class="reference external" href="https://samtools.github.io/bcftools/bcftools.html">https://samtools.github.io/bcftools/bcftools.html</a></p>
<p>We put some examples here you might find helpful using the <code class="docutils literal notranslate"><span class="pre">bcftools</span> <span class="pre">query</span></code> utility, which can output information from VCF files in customized text formats, and also allows for a rich set of options to filter data by sample, position, VCF fields, etc.</p>
<ul class="simple">
<li><p>Extract genotypes for a single sample (e.g. NA12878) in tab-delimited format</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>VCF=~/public/1000Genomes/ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz
bcftools query -s NA12878 -f &#39;%CHROM\t%POS\t[%GT\n]&#39; $VCF
</pre></div>
</div>
<ul class="simple">
<li><p>Count the number of singleton variants in this sample, restricting to bi-allelic sites. A singleton is a variant for which only a single copy of the alternate allele was seen in the entire cohort. The command below uses the <code class="docutils literal notranslate"><span class="pre">-i</span></code> option of bcftools to restrict to sites where the number of alternate alleles is 1 and the alternate allele count is 1. See more on <a class="reference external" href="https://samtools.github.io/bcftools/bcftools.html#expressions">bcftools expressions</a>.</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>bcftools query --include &quot;N_ALT=1 &amp;&amp; INFO/AC=1&quot; \
  -s NA12878 -f &#39;[%GT\n]&#39; $VCF | \
  grep -v &quot;0|0&quot; | wc -l
</pre></div>
</div>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./ancestry/plink1kg"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="../plink_1000G.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Tutorial 2: Working with 1000 Genomes data and plink</p>
      </div>
    </a>
    <a class="right-next"
       href="plink.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">T2.2 Some tips and examples for using plink</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#overview">Overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#genomes-population-codes">1000 Genomes population codes</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#genomes-data-vcfs">1000 Genomes data - VCFs</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#genomes-data-populations">1000 Genomes data - Populations</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#using-1000-genomes-data-in-plink">Using 1000 Genomes data in plink</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#wrangling-1000-genomes-vcf-files-with-bcftools">Wrangling 1000 Genomes VCF files with bcftools</a></li>
</ul>
</li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Melissa Gymrek
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549"></script>
<script defer src="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>