

<!DOCTYPE html>


<html lang="en" data-content_root="" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>T2.2 Some tips and examples for using plink &#8212; CSE284 - Personal Genomics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="../../_static/styles/bootstrap.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=5b4479735964841361fd" rel="stylesheet" />

  
  <link href="../../_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=5b4479735964841361fd" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css" />
    <link rel="stylesheet" href="../../_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=5b4479735964841361fd" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd" />
  <script src="../../_static/vendor/fontawesome/6.1.2/js/all.min.js?digest=5b4479735964841361fd"></script>

    <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/clipboard.min.js"></script>
    <script src="../../_static/copybutton.js"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../../_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'ancestry/plink1kg/plink';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Chapter XX: Introduction to complex traits" href="../../complextraits/overview.html" />
    <link rel="prev" title="T2.1 The 1000 Genomes Project dataset (and wrangling VCF files)" href="1000G.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <div id="pst-scroll-pixel-helper"></div>

  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>
    Back to top
  </button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  

<a class="navbar-brand logo" href="../../index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../_static/logo.png" class="logo__image only-light" alt="CSE284 - Personal Genomics - Home"/>
    <script>document.write(`<img src="../../_static/logo.png" class="logo__image only-dark" alt="CSE284 - Personal Genomics - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../index.html">
                    Personal Genomics for Bioinformaticians
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 1 - Introduction</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/intro_to_your_genome.html">Chapter 1: Introduction to the human genome</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-1"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/structure_DNA.html">1.1 The structure of DNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/genome_organization.html">1.2 Organization of the human genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/central_dogma.html">1.3 Genes and the central dogma</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/noncoding.html">1.4 The non-coding genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/review_questions.html">1.5 Review Questions</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/intro_genome_tech.html">Chapter 2: Overview of technologies for genome analysis</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-2"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/genotype_arrays.html">2.1 Genotyping arrays</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/ngs.html">2.2 Next-generation sequencing</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/long_reads.html">2.3 Long-read technologies</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/tech_summary.html">2.4 Summary of genomics technologies</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/review_questions.html">2.5 Review Questions</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/genome_file_formats.html">Tutorial 1: file formats for describing genetic variation</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-3"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/vcf.html">T1.1 VCF files</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/plink.html">T1.2 Plink files</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/pgen.html">T1.2 PGEN files</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 2 - Ancestry</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../intro_popgen.html">Chapter 3: Introduction to population genetics</a><input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-4"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../popgen/genetic_variation_terminology.html">3.1 Genetic variation terminology</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/forces_evolution.html">3.2 The four forces of evolution</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/mutations.html">3.3 Types and patterns of mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/drift.html">3.4 Genetic drift</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/selection.html">3.5 Selection</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/hwe.html">3.6 Hardy-Weinberg Equilibrium and random mating</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/fst.html">3.7 Measuring population differentiation with Fst</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/recombination.html">3.8 Recombination</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/review_questions.html">3.9 Review Questions</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../global_ancestry.html">Chapter 4: Global ancestry</a><input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-5"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../global/pca_global_ancestry.html">2.3 Global ancestry analysis with PCA</a></li>


<li class="toctree-l2"><a class="reference internal" href="../global/admixture_global_ancestry.html">4.2 Global ancestry analysis with ADMIXTURE</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../relative_finding.html">Chapter 5: Relative finding</a><input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-6"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd.html">5.1 Identity by descent</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd_pedigree.html">5.2 Expected IBD of close relatives</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd_plink.html">5.3 Computing expected IBD - plink method</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd_segments.html">5.4 Computing expected IBD segment sharing</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/genetic_genealogy.html">5.5 Genetic genealogy</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../admixture.html">Chapter 6: Admixture</a></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../plink_1000G.html">Tutorial 2: Working with 1000 Genomes data and plink</a><input checked="" class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-7"><i class="fa-solid fa-chevron-down"></i></label><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="1000G.html">T2.1 The 1000 Genomes Project dataset (and wrangling VCF files)</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">T2.2 Some tips and examples for using plink</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 3 - Complex traits</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../complextraits/overview.html">Chapter XX: Introduction to complex traits</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../complextraits/gwas.html">Chapter XX: Genome-wide association studies</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../complextraits/lmm.html">Chapter XX: Linear mixed models</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../complextraits/qtls.html">Chapter XX: Quantitative trait loci and functional genomics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../complextraits/fine-mapping.html">Chapter XX: Fine-mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../complextraits/prs.html">Chapter XX: Polygenic risk scores</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 4 - Trans-ancestry approaches</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../multianc/fine-mapping.html">Chapter XX: PRS and portability</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../multianc/prs.html">Chapter XX: PRS and portability</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 5 - Beyond common variants</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../beyond/rare.html">Chapter XX: Rare variants</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../beyond/complex.html">Chapter XX: Complex variants</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 6 - Ethics</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../ethics/privacy.html">Chapter XX: Ethics part I - privacy</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../ethics/health.html">Chapter XX: Ethics part II - health</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../ethics/equity.html">Chapter XX: Ethics part I - equity</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Appendix 1 - review of math+CS concepts</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../appendix/linear_algebra.html">Chapter A1.1 Review of linear algebra concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../appendix/hmm.html">Chapter A2.2: Hidden Markov Models</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Appendix 2 - glossary and notation</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../appendix/unix.html">Chapter A2.1: Unix commands used</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../appendix/software.html">Chapter A2.2: List of software tools used</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../appendix/glossary.html">Chapter A2.3: Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../appendix/notation.html">Chapter A2.4: Notation used throughout the book</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/gymrek-lab/personal-genomics-textbook" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/gymrek-lab/personal-genomics-textbook/issues/new?title=Issue%20on%20page%20%2Fancestry/plink1kg/plink.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/ancestry/plink1kg/plink.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>T2.2 Some tips and examples for using plink</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#the-anatomy-of-a-plink-command">The anatomy of a plink command</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#specifying-input-genotypes">Specifying input genotypes</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#converting-between-genotype-formats">Converting between genotype formats</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#caveats-when-dealing-with-vcf-files">Caveats when dealing with VCF files</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#example-plink-use-case-ld-pruning">Example plink use case - LD pruning</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="t2-2-some-tips-and-examples-for-using-plink">
<h1>T2.2 Some tips and examples for using plink<a class="headerlink" href="#t2-2-some-tips-and-examples-for-using-plink" title="Permalink to this heading">#</a></h1>
<p><a class="reference external" href="https://www.cog-genomics.org/plink/1.9/">Plink</a> contains implementation of many different methods and utilities that can be performed on genetic data. There are many different options, which can make it a bit confusing if you’ve never used it before. This brief tutorial goes over some plink basics, and through some example use cases.</p>
<p><strong>Note: when using plink, things will run much faster if you use the high memory instances on datahub.</strong></p>
<section id="the-anatomy-of-a-plink-command">
<h2>The anatomy of a plink command<a class="headerlink" href="#the-anatomy-of-a-plink-command" title="Permalink to this heading">#</a></h2>
<p>While there are many different ways to use plink, almost all commands will take the following form:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">plink</span> \
  <span class="o">&lt;</span><span class="n">option</span> <span class="n">to</span> <span class="n">specify</span> <span class="nb">input</span> <span class="n">genotypes</span><span class="o">&gt;</span> \
  <span class="o">--</span><span class="n">out</span> <span class="n">outprefix</span> \
  <span class="p">[</span><span class="n">command</span><span class="p">]</span>
  <span class="p">[</span><span class="n">other</span> <span class="n">options</span><span class="p">]</span>
</pre></div>
</div>
<p>That is, you must specify:</p>
<ul class="simple">
<li><p>The path to the genotype information for your samples. This argument can take multiple possible forms (see below)</p></li>
<li><p>A prefix to name output files. e.g. <code class="docutils literal notranslate"><span class="pre">--out</span> <span class="pre">prefix</span></code> will name files <code class="docutils literal notranslate"><span class="pre">prefix.log</span></code>, etc. The exact output files depend on the command used.</p></li>
<li><p>A command to run. e.g. <code class="docutils literal notranslate"><span class="pre">--pca</span></code> or <code class="docutils literal notranslate"><span class="pre">--genome</span></code>. Every run has to have some kind of command, even if it is just a command to output a certain file format, like <code class="docutils literal notranslate"><span class="pre">--make-bed</span></code> or <code class="docutils literal notranslate"><span class="pre">--recode</span></code>.</p></li>
</ul>
<p>Finally, you may wish to specify other options, e.g. to restrict analysis to a certain chromosome, filter SNPs by MAF, etc.</p>
<p>Below we talk more about these different arguments.</p>
<section id="specifying-input-genotypes">
<h3>Specifying input genotypes<a class="headerlink" href="#specifying-input-genotypes" title="Permalink to this heading">#</a></h3>
<p>For almost any use case, plink will need access to the actual genotypes (e.g. 0s, 1s, and 2s for each variant) for one or more samples. There are multiple ways to provide genotypes as input to plink. These include:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">--vcf</span> <span class="pre">&lt;file.vcf&gt;</span></code>: Plink can directly read VCF files. They can be either bgzipped or not. This is convenient since many big datasets (e.g. 1000Genomes) are in VCF format already. But there are some caveats (see below).</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--file</span> <span class="pre">&lt;prefix&gt;</span></code>: The original plink format has several files per dataset: <code class="docutils literal notranslate"><span class="pre">$prefix.ped</span></code> gives family/sample information and genotype calls, and <code class="docutils literal notranslate"><span class="pre">$prefix.map</span></code> specifies the locations, alleles, and IDs of the SNPs. <code class="docutils literal notranslate"><span class="pre">--file</span> <span class="pre">prefix</span></code> tells plink to look for <code class="docutils literal notranslate"><span class="pre">prefix.map</span></code> and <code class="docutils literal notranslate"><span class="pre">prefix.ped</span></code>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--bfile</span> <span class="pre">&lt;prefix&gt;</span></code>: This is a binary (compressed) version of the original plink format. Plink will look for <code class="docutils literal notranslate"><span class="pre">prefix.bed</span></code> (compressed genotypes) and <code class="docutils literal notranslate"><span class="pre">prefix.bim</span></code> (SNP info in a simple text file)</p></li>
</ul>
<p>Plink handles many other formats, but all of our data for the course will come in one of the formats above. Read more about plink file formats here: <a class="reference external" href="https://www.cog-genomics.org/plink/1.9/formats">https://www.cog-genomics.org/plink/1.9/formats</a>.</p>
<section id="converting-between-genotype-formats">
<h4>Converting between genotype formats<a class="headerlink" href="#converting-between-genotype-formats" title="Permalink to this heading">#</a></h4>
<p>You can use plink to convert genotypes from one format to another. e.g.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>VCF=/datasets/cs284-sp22-a00-public/1000Genomes/ALL.chr21.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz

# Convert VCF to ped/map
# (using only couple of variants to make this smaller)
plink \
 --vcf $VCF \
 --recode \
 --out test \
 --from rs559462325 --to rs565663130

# Convert ped/map to bed/bim
plink \
 --file test \
 --make-bed \
 --out test

# Convert bed/bim to VCF
plink \
 --bfile test \
 --recode vcf bgz \
 --out test
</pre></div>
</div>
</section>
<section id="caveats-when-dealing-with-vcf-files">
<h4>Caveats when dealing with VCF files<a class="headerlink" href="#caveats-when-dealing-with-vcf-files" title="Permalink to this heading">#</a></h4>
<p>Plink currently loses some information present in VCF files, which can be a common source of errors and confusion. You shouldn’t need to deal with most of these things (except the <code class="docutils literal notranslate"><span class="pre">--double-id</span></code> issue) in your problem sets but they are good to be aware of:</p>
<ul class="simple">
<li><p>It doesn’t properly deal with multi-allelic variants (<a class="reference external" href="https://www.cog-genomics.org/plink/1.9/data#merge3">https://www.cog-genomics.org/plink/1.9/data#merge3</a>)</p></li>
<li><p>Plink traditionally included both family and sample IDs for each sample. But VCFs only have sample IDs. To deal with this, you’ll have to use the <code class="docutils literal notranslate"><span class="pre">--double-id</span></code> option if you want to extract sets of samples. See the 1000 Genomes tutorial from week 1.</p></li>
<li><p>If you convert from VCF to another plink format, it order alleles based on minor vs. major allele. This can be problematic and confusing, since VCFs have a strict notion of reference allele (what is in the reference genome) and alternate allele. See option <code class="docutils literal notranslate"><span class="pre">--keep-allele-order</span></code>.</p></li>
<li><p>Phase data is ignored (and orders might be switched in output files).</p></li>
</ul>
</section>
</section>
</section>
<section id="example-plink-use-case-ld-pruning">
<h2>Example plink use case - LD pruning<a class="headerlink" href="#example-plink-use-case-ld-pruning" title="Permalink to this heading">#</a></h2>
<p>Some analyses, such as PCA or running ADMIXTURE, assume we have a set of SNPs in linkage <em>equilibrium</em>. In those cases, it is common to “prune” SNPs to remove highly correlated SNPs from analysis. See the example below for steps to do this.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>VCF=/datasets/cs284-sp22-a00-public/1000Genomes/ALL.chr21.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz

# Get a list of independent SNPs
# --indep-pairwise &lt;window size&gt; &lt;step size&gt; &lt;r2 threshold&gt;
# Window size = 50 variants
# Step size = variant count to shift the window at the end of each step
# r2 threshold = 0.1. at each step, pairs of variants in the current window with 
# squared correlation greater than the threshold are noted, and variants are 
# greedily pruned from the window until no such pairs remain
# Outputs prune-example.prune.in (variants to keep) and prune-example.prune.out (variants to prune)
plink --vcf $VCF --indep-pairwise 50 10 0.1 --out prune-example

# Extract a new dataset with only those SNPs
plink --vcf $VCF --extract prune-example.in --out prune-example.pruned --make-bed
</pre></div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./ancestry/plink1kg"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="1000G.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">T2.1 The 1000 Genomes Project dataset (and wrangling VCF files)</p>
      </div>
    </a>
    <a class="right-next"
       href="../../complextraits/overview.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Chapter XX: Introduction to complex traits</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#the-anatomy-of-a-plink-command">The anatomy of a plink command</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#specifying-input-genotypes">Specifying input genotypes</a><ul class="nav section-nav flex-column">
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#converting-between-genotype-formats">Converting between genotype formats</a></li>
<li class="toc-h4 nav-item toc-entry"><a class="reference internal nav-link" href="#caveats-when-dealing-with-vcf-files">Caveats when dealing with VCF files</a></li>
</ul>
</li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#example-plink-use-case-ld-pruning">Example plink use case - LD pruning</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Melissa Gymrek
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2022.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../_static/scripts/bootstrap.js?digest=5b4479735964841361fd"></script>
<script src="../../_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>