
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>3.7 Recombination &#8212; CSE284 - Personal Genomics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!-- 
    this give us a css class that will be invisible only if js is disabled 
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css?v=be8a1c11" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- So that users can add custom icons -->
  <script src="../../_static/scripts/fontawesome.js?digest=26a4bc78f4c0ddb94549"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549" />

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'ancestry/popgen/recombination';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Activity: Exploring the Wright-Fisher model of genetic drift" href="WF-simulations.html" />
    <link rel="prev" title="3.6 Measuring population differentiation with Fst" href="fst.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="" />
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../_static/logo.png" class="logo__image only-light" alt="CSE284 - Personal Genomics - Home"/>
    <img src="../../_static/logo.png" class="logo__image only-dark pst-js-only" alt="CSE284 - Personal Genomics - Home"/>
  
  
</a></div>
        <div class="sidebar-primary-item">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../index.html">
                    Personal Genomics for Bioinformaticians
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 1 - Introduction</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/intro_to_your_genome.html">Chapter 1: Introduction to the human genome</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/structure_DNA.html">1.1 The structure of DNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/genome_organization.html">1.2 Organization of the human genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/central_dogma.html">1.3 Genes and the central dogma</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/noncoding.html">1.4 The non-coding genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/genetic_variation_terminology.html">1.5 Genetic variation terminology</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/intro_genome_tech.html">Chapter 2: Overview of technologies for genome analysis</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/genotype_arrays.html">2.1 Genotyping arrays</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/ngs.html">2.2 Next-generation sequencing</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/long_reads.html">2.3 Long-read technologies</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/tech_summary.html">2.4 Summary of genomics technologies</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/genome_file_formats.html">Tutorial 1: file formats for describing genetic variation</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/vcf.html">T1.2 VCF files</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/plink.html">T1.2 Plink files</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 2 - Ancestry</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../intro_popgen.html">Chapter 3: Introduction to population genetics</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="forces_evolution.html">3.1 The four forces of evolution</a></li>
<li class="toctree-l2"><a class="reference internal" href="mutations.html">3.2 Types and patterns of mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="drift.html">3.3 Genetic drift</a></li>
<li class="toctree-l2"><a class="reference internal" href="selection.html">3.4 Selection</a></li>
<li class="toctree-l2"><a class="reference internal" href="hwe.html">3.5 Hardy-Weinberg Equilibrium and random mating</a></li>
<li class="toctree-l2"><a class="reference internal" href="fst.html">3.6 Measuring population differentiation with Fst</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">3.7 Recombination</a></li>
<li class="toctree-l2"><a class="reference internal" href="WF-simulations.html">Activity: Exploring the Wright-Fisher model of genetic drift</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../global_ancestry.html">Chapter 4: Global ancestry</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../global/pca_global_ancestry.html">4.1 Global ancestry analysis with PCA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../global/admixture_global_ancestry.html">4.2 Global ancestry analysis with ADMIXTURE</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../relative_finding.html">Chapter 5: Relative finding</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd.html">5.1 Identity by descent</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd_pedigree.html">5.2 Expected IBD of close relatives</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd_plink.html">5.3 Computing expected IBD - plink method</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd_segments.html">5.4 Computing expected IBD segment sharing</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../plink_1000G.html">Tutorial 2: Working with 1000 Genomes data and plink</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../plink1kg/1000G.html">T2.1 The 1000 Genomes Project dataset (and wrangling VCF files)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../plink1kg/plink.html">T2.2 Some tips and examples for using plink</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 3 - From genotypes to phenotypes</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../../complextraits/gwas.html">Chapter 7: GWAS for complex traits</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/quantitative_traits.html">7.1 GWAS for quantitative traits</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/GWAS-simulations.html">7.2 Exploring GWAS association statistics</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/confounding.html">7.3 Confounding factors in GWAS</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/case_control_traits.html">7.4 GWAS for case-control traits</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/lmms.html">7.5 Linear mixed models</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/gwas/power.html">7.6 Power to detect associations</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../complextraits/heritability.html">Chapter 8: Heritability</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/heritability/relatives.html">8.1 Measuring heritability in related individuals</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/heritability/lmms-h2.html">8.2 Measuring SNP-based heritability using LMMs</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../complextraits/prs.html">Chapter 9: Polygenic risk scores</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/prs/clumping-thresholding.html">9.1 Clumping + Threshold (C+T) method</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../complextraits/prs/bayesian_prs.html">9.2 Bayesian methods for polygenic risk scores</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Appendix 1 - glossary and notation</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../appendix/unix.html">Chapter A1.1: Unix cheat sheet</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/gymrek-lab/personal-genomics-textbook" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/gymrek-lab/personal-genomics-textbook/issues/new?title=Issue%20on%20page%20%2Fancestry/popgen/recombination.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/ancestry/popgen/recombination.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button>


<button class="btn btn-sm pst-navbar-icon search-button search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
</button>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>3.7 Recombination</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mitosis-vs-meiosis">Mitosis vs. meiosis</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#recombination-results-in-new-combinations-of-alleles">Recombination results in new combinations of alleles</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#linkage-disequilibrium">Linkage disequilibrium</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#d-and-d-prime">D and D prime</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#r-2"><span class="math notranslate nohighlight">\(r^2\)</span></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#ld-patterns">LD patterns</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="recombination">
<h1>3.7 Recombination<a class="headerlink" href="#recombination" title="Link to this heading">#</a></h1>
<p>We previously discussed how <em>mutations</em> contribute new genetic variants as fuel for evolution. Here we discuss another process, <em>recombination</em>, in which diploid genome copies exchange genetic material, which can result in new combinations of variants together on the same chromosome copy. Recombination is an important additional force driving changes in genomes and patterns of genetic variation over time.</p>
<section id="mitosis-vs-meiosis">
<h2>Mitosis vs. meiosis<a class="headerlink" href="#mitosis-vs-meiosis" title="Link to this heading">#</a></h2>
<p>Most cells in the human body are <em>diploid</em>, meaning they each have to copies of each chromosome (one copy from each of your parents). These diploid cells usually divide by a process known as <em>mitosis</em>, in which the entire diploid genome is copied, and a single cell splits into two daughter cells, each with an identical diploid genome to that of the original cell. (Note: they might not quite be identical. During cell division of non-germ cells, <em>somatic mutations</em> may arise. We are going to largely ignore somatic mutations in this course but there is growing appreciation of the contribution of somatic mutations to human health.)</p>
<p>On the other hand, <em>gametes</em> (egg and seprm) are <em>haploid</em>, meaning they have only a single copy of each chromosome. These are generated by a different type of cell division known as <em>meiosis</em>. First, similar to mitosis, a diploid cell undergoes DNA replication to make a new copy of the genome. Then, there are two separate rounds of cell division (meiosis I and meiosis II) to ultimately generate four <em>haploid</em> gamete cells.</p>
<p><img alt="meiosis" src="../../_images/meiosis.png" /></p>
<p>After DNA replication but before the first division, <em>homologous recombination</em> may occur. In the figure above, black and red denote the two original copies of the genome. After recombination occurs, we still have two copies of the genome, but there has been an exchange of genetic material. The resulting chromosome copies are mosaics of the two original genome copies (red and black).</p>
<p>This has important implications when we think about inheritance of genome copies. For example, this means the copy of the chromosome you received from your mom is not identical that in either you maternal grandmother or maternal grandfather, but rather is more likely a mosaic containing material from both of your maternal grandparents (and same idea for the genome copy you received from your father).</p>
<p>There is around one recombination event per chromosome per generation. The total number of recombination events across all chromosomes per generation is around 35.</p>
</section>
<section id="recombination-results-in-new-combinations-of-alleles">
<h2>Recombination results in new combinations of alleles<a class="headerlink" href="#recombination-results-in-new-combinations-of-alleles" title="Link to this heading">#</a></h2>
<p>Another important implication of recombination is that can result in new combinations of alleles. Let’s illustrate that process below.</p>
<p>To do so, we’ll introduce a new term. A <em>haplotype</em> will refer to a combination of alleles found together (and typically nearby) on the same copy of the chromosome. Consider two positions which start out as “A” and “C” on an ancestral copy of a haplotype (black haplotype below). At some point, the “A” mutates to a “T” (generating the red haplotype below). Then later the “C” at the second position on the red haplotype mutates to a “G” to create a new haplotype (orange haplotype below). In the absence of recombination, we would never observe an “A” at position 1 together with a “G” at position 2.</p>
<p><img alt="meiosis" src="../../_images/recomb1.png" /></p>
<p>Now, consider that the black and the orange chromosomes undergo recombination. This can make a new combination of variants (“A” and “G”) on the same chromosome copy that otherwise impossible to obtain:</p>
<p><img alt="meiosis" src="../../_images/recomb2.png" /></p>
<p>This process of recombination is happening each generation, making new arrangements of variants. It is therefore an important driving force, alongside mutations themselves, of genome evolution.</p>
<p>Notably, recombination does not occur completely at random. Rather, there are “hotspots” in the genome where breakpoints tend to occur. In some cases, these hotspots are driven by special sequence “motifs” in the genome which may encode for regulatory proteins such as <a class="reference external" href="https://www.science.org/doi/10.1126/science.1183439"><em>PRDM9</em></a>. The presence of these hotspots will create unique patterns of correlation across the genome which we will explore below.</p>
</section>
<section id="linkage-disequilibrium">
<h2>Linkage disequilibrium<a class="headerlink" href="#linkage-disequilibrium" title="Link to this heading">#</a></h2>
<p>Generally, two variants that are very close together on a chromosome are less likely to be broken up by recombination than those that are very far away. As a consequence, genotypes for variants that are nearby are likely to be correlated, or in <em>linkage disequilibrium (LD)</em> with each other. We will discuss several metrics to compute the LD between pairs of variants.</p>
<p>Consider two SNPs, SNP1 and SNP2, with MAFs <span class="math notranslate nohighlight">\(q_1\)</span> and <span class="math notranslate nohighlight">\(q_2\)</span>. We can go observe many haplotypes in the population, and count how many times we saw each combination of alleles at these two positions. e.g. <span class="math notranslate nohighlight">\(p_{00}\)</span> denotes the percent of haplotypes that have the major allele at both positions:</p>
<p><img alt="ld" src="../../_images/ld.png" /></p>
<p>We consider the two SNPs to be in <em>linkage equilibrium</em> (uncorrelated) if the chance to observe a particular combination of alleles is the same as if we had drawn each allele independently, i.e.:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(p_{00} = (1-q_1)(1-q_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(p_{10} = q_1(1-q_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(p_{01} = (1-q_1)q_2\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(p_{11} = q_1q_2\)</span></p></li>
</ul>
<p>On the other hand, that are in LD if these relationships are not true:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(p_{00} \neq (1-q_1)(1-q_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(p_{10} \neq q_1(1-q_2)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(p_{01} \neq (1-q_1)q_2\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(p_{11} \neq q_1q_2\)</span></p></li>
</ul>
<section id="d-and-d-prime">
<h3>D and D prime<a class="headerlink" href="#d-and-d-prime" title="Link to this heading">#</a></h3>
<p>One way we can quantify LD is with the <span class="math notranslate nohighlight">\(D\)</span> metric:</p>
<div class="math notranslate nohighlight">
\[ D = p_{11} - q_1q_2 \]</div>
<p>This measures how far the observed haplotype frequencies deviate from the expected. If there is no LD, then this quantity will be 0. Note, the sign is arbitrary. A limitation of this metric is that the range depends on the MAFs of SNP1 and SNP2. Therefore a more commonly used metric is <span class="math notranslate nohighlight">\(D'\)</span>, which normalizes by the maximum possible value of <span class="math notranslate nohighlight">\(D\)</span>:</p>
<div class="math notranslate nohighlight">
\[ D' = D/D_{max}\]</div>
<p>where <span class="math notranslate nohighlight">\(D_{max} = \max\{-q_1q_2, -(1-q_1)(1-q_2)\}\)</span> if <span class="math notranslate nohighlight">\(D&lt;0\)</span> and <span class="math notranslate nohighlight">\(D_{max} = \min\{q_1(1-q_2), (1-q_1)q_2\} if \)</span>D&gt;0$.</p>
</section>
<section id="r-2">
<h3><span class="math notranslate nohighlight">\(r^2\)</span><a class="headerlink" href="#r-2" title="Link to this heading">#</a></h3>
<p>Another way to measure LD is using squared Pearson correlation between the alleles at SNP1 vs. SNP2 across many haplotypes in the population. This comes out to:</p>
<div class="math notranslate nohighlight">
\[r^2 = \frac{D^2}{q_1(1-q_1)q_2(1-q_2)}\]</div>
<p><span class="math notranslate nohighlight">\(r^2\)</span> is between 0 and 1 (although it is not always possible for <span class="math notranslate nohighlight">\(r^2\)</span> to reach 1, depending on the allele frequencies of the two SNPs considered). It will also be a convenient metric later on when we talk about GWAS and fine-mapping.</p>
</section>
</section>
<section id="ld-patterns">
<h2>LD patterns<a class="headerlink" href="#ld-patterns" title="Link to this heading">#</a></h2>
<p>Below shows a representative example region of the genome, in which the LD (<span class="math notranslate nohighlight">\(r^2\)</span>) is plotted for each pair of SNPs.</p>
<p><img alt="ld_hotspots" src="../../_images/ld_hotspots.png" /></p>
<p>Notice the triangle shape. The triangles indicate large blocks (<em>LD blocks</em>) of variants that are in high correlation with each other. The breaks between triangles indicate <em>recombination hotspots</em>, where recombination frequently occurs and has broken down the correlation between variants. Note, LD patterns often vary substantially by ancestry.</p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./ancestry/popgen"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="fst.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">3.6 Measuring population differentiation with Fst</p>
      </div>
    </a>
    <a class="right-next"
       href="WF-simulations.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Activity: Exploring the Wright-Fisher model of genetic drift</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#mitosis-vs-meiosis">Mitosis vs. meiosis</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#recombination-results-in-new-combinations-of-alleles">Recombination results in new combinations of alleles</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#linkage-disequilibrium">Linkage disequilibrium</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#d-and-d-prime">D and D prime</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#r-2"><span class="math notranslate nohighlight">\(r^2\)</span></a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#ld-patterns">LD patterns</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Melissa Gymrek
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549"></script>
<script defer src="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>