

<!DOCTYPE html>


<html lang="en" data-content_root="" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>7.1 GWAS for quantitative traits &#8212; CSE284 - Personal Genomics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=8d27b9dea8ad943066ae" rel="stylesheet" />
<link href="../../_static/styles/bootstrap.css?digest=8d27b9dea8ad943066ae" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=8d27b9dea8ad943066ae" rel="stylesheet" />

  
  <link href="../../_static/vendor/fontawesome/6.5.1/css/all.min.css?digest=8d27b9dea8ad943066ae" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.1/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.1/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.1/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css" />
    <link rel="stylesheet" href="../../_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=8d27b9dea8ad943066ae" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=8d27b9dea8ad943066ae" />
  <script src="../../_static/vendor/fontawesome/6.5.1/js/all.min.js?digest=8d27b9dea8ad943066ae"></script>

    <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/clipboard.min.js"></script>
    <script src="../../_static/copybutton.js"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../../_static/sphinx-thebe.js"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'complextraits/gwas/quantitative_traits';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="7.2 Exploring GWAS association statistics" href="GWAS-simulations.html" />
    <link rel="prev" title="Chapter 7: GWAS for complex traits" href="../gwas.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a id="pst-skip-link" class="skip-link" href="#main-content">Skip to main content</a>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>
    Back to top
  </button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <header class="bd-header navbar navbar-expand-lg bd-navbar">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  

<a class="navbar-brand logo" href="../../index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../_static/logo.png" class="logo__image only-light" alt="CSE284 - Personal Genomics - Home"/>
    <script>document.write(`<img src="../../_static/logo.png" class="logo__image only-dark" alt="CSE284 - Personal Genomics - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../index.html">
                    Personal Genomics for Bioinformaticians
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 1 - Introduction</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/intro_to_your_genome.html">Chapter 1: Introduction to the human genome</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-1"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/structure_DNA.html">1.1 The structure of DNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/genome_organization.html">1.2 Organization of the human genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/central_dogma.html">1.3 Genes and the central dogma</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/noncoding.html">1.4 The non-coding genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/genetic_variation_terminology.html">1.5 Genetic variation terminology</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/intro_genome_tech.html">Chapter 2: Overview of technologies for genome analysis</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-2"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/genotype_arrays.html">2.1 Genotyping arrays</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/ngs.html">2.2 Next-generation sequencing</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/long_reads.html">2.3 Long-read technologies</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/tech_summary.html">2.4 Summary of genomics technologies</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/genome_file_formats.html">Tutorial 1: file formats for describing genetic variation</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-3"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/vcf.html">T1.2 VCF files</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/plink.html">T1.2 Plink files</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 2 - Ancestry</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../../ancestry/intro_popgen.html">Chapter 3: Introduction to population genetics</a><input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-4"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/popgen/forces_evolution.html">3.1 The four forces of evolution</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/popgen/mutations.html">3.2 Types and patterns of mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/popgen/drift.html">3.3 Genetic drift</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/popgen/selection.html">3.4 Selection</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/popgen/hwe.html">3.5 Hardy-Weinberg Equilibrium and random mating</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/popgen/fst.html">3.6 Measuring population differentiation with Fst</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/popgen/recombination.html">3.7 Recombination</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/popgen/WF-simulations.html">Activity: Exploring the Wright-Fisher model of genetic drift</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../ancestry/global_ancestry.html">Chapter 4: Global ancestry</a><input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-5"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/global/pca_global_ancestry.html">4.1 Global ancestry analysis with PCA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/global/admixture_global_ancestry.html">4.2 Global ancestry analysis with ADMIXTURE</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../ancestry/relative_finding.html">Chapter 5: Relative finding</a><input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-6"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/relfind/ibd.html">5.1 Identity by descent</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/relfind/ibd_pedigree.html">5.2 Expected IBD of close relatives</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/relfind/ibd_plink.html">5.3 Computing expected IBD - plink method</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/relfind/ibd_segments.html">5.4 Computing expected IBD segment sharing</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../ancestry/plink_1000G.html">Tutorial 2: Working with 1000 Genomes data and plink</a><input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-7"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/plink1kg/1000G.html">T2.1 The 1000 Genomes Project dataset (and wrangling VCF files)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ancestry/plink1kg/plink.html">T2.2 Some tips and examples for using plink</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 3 - From genotypes to phenotypes</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../gwas.html">Chapter 7: GWAS for complex traits</a><input checked="" class="toctree-checkbox" id="toctree-checkbox-8" name="toctree-checkbox-8" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-8"><i class="fa-solid fa-chevron-down"></i></label><ul class="current">
<li class="toctree-l2 current active"><a class="current reference internal" href="#">7.1 GWAS for quantitative traits</a></li>
<li class="toctree-l2"><a class="reference internal" href="GWAS-simulations.html">7.2 Exploring GWAS association statistics</a></li>
<li class="toctree-l2"><a class="reference internal" href="confounding.html">7.3 Confounding factors in GWAS</a></li>
<li class="toctree-l2"><a class="reference internal" href="case_control_traits.html">7.4 GWAS for case-control traits</a></li>
<li class="toctree-l2"><a class="reference internal" href="lmms.html">7.5 Linear mixed models</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../heritability.html">Chapter 8: Heritability</a><input class="toctree-checkbox" id="toctree-checkbox-9" name="toctree-checkbox-9" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-9"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../heritability/relatives.html">8.1 Measuring heritability in related individuals</a></li>
<li class="toctree-l2"><a class="reference internal" href="../heritability/lmms-h2.html">8.2 Measuring SNP-based heritability using LMMs</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Appendix 1 - glossary and notation</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../appendix/unix.html">Chapter A1.1: Unix cheat sheet</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/gymrek-lab/personal-genomics-textbook" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/gymrek-lab/personal-genomics-textbook/issues/new?title=Issue%20on%20page%20%2Fcomplextraits/gwas/quantitative_traits.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/complextraits/gwas/quantitative_traits.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>7.1 GWAS for quantitative traits</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#gwas-overview">7.1.2 GWAS Overview</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#association-testing-for-a-quantitative-trait">7.1.3 Association testing for a quantitative trait</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#visualizing-gwas-results">7.1.4 Visualizing GWAS results</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#gwas-significance-threshold">7.1.5 GWAS significance threshold</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#more-on-gwas-qc">7.1.6 More on GWAS QC</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="gwas-for-quantitative-traits">
<h1>7.1 GWAS for quantitative traits<a class="headerlink" href="#gwas-for-quantitative-traits" title="Permalink to this heading">#</a></h1>
<p>In this section we introduce the technique of <em>genome-wide association studies</em> (GWAS), focusing first on quantitative traits. A subsequent section will describe GWAS for case-control traits.</p>
<p>The major goal of GWAS is to identify regions of the genome associated with a particular trait. Since we will be focusing on <em>complex traits</em>, in most cases we will find multiple, sometimes thousands, of trait-associated regions.</p>
<section id="gwas-overview">
<h2>7.1.2 GWAS Overview<a class="headerlink" href="#gwas-overview" title="Permalink to this heading">#</a></h2>
<p>The starting point for our GWAS will be a large cohort of samples for which we have both:</p>
<ul class="simple">
<li><p><em>Phenotypes</em>: this is the measurement of the trait of interest we are studying. For example, a person’s height.</p></li>
<li><p><em>Genotypes</em>: typically we would be working with genome-wie SNP genotypes (0s, 1s, and 2s), which is the same type of data we worked with in our ancestry module. Genotypes may be obtained either from SNP genotyping arrays or from whole genome sequencing. If genotypes were obtained from arrays, typically there is a round of imputation to impute SNPs that were not directly genotyped. More on that later.</p></li>
</ul>
<p>Note, the cohorts collected for GWAS can be massive. Early GWAS cohorts consisted of several thousand individuals. Recent biobanks have hundreds of thousands of samples. The <a class="reference external" href="https://www.nature.com/articles/s41586-022-05275-y">most recent height GWAS</a> had around 5 million individuals! Such large sample sizes are needed to have power to detect variants that have low effect sizes or are not very common in the population. We’ll explore power in a later section.</p>
<p>Once the cohort is collected and undergoes careful quality control to filter problematic samples or variants, the main step of GWAS is to perform an association test for each variant. The results of these tests will point us to the regions of the genome most strongly associated with the trait. The next step gives more details for how this is done.</p>
</section>
<section id="association-testing-for-a-quantitative-trait">
<h2>7.1.3 Association testing for a quantitative trait<a class="headerlink" href="#association-testing-for-a-quantitative-trait" title="Permalink to this heading">#</a></h2>
<p>We will perform a separate association test for each SNP. Note, we are implicitly making an assumption here that the effect of each variant is <em>independent</em> of all others and works in an <em>additive</em> fashion.</p>
<p>Let:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(Y = \{y_1, ... y_n\}\)</span> be the phenotype values for each sample <span class="math notranslate nohighlight">\(1..n\)</span>.</p></li>
<li><p><span class="math notranslate nohighlight">\(X_j = \{x_{ij}\}; i=1..n; j=1..m\)</span> be the genotype values for SNP <span class="math notranslate nohighlight">\(j\)</span>. <span class="math notranslate nohighlight">\(x_{ij}\)</span> denotes the genotype (0, 1, or 2 copies of the alternate allele) for sample <span class="math notranslate nohighlight">\(i\)</span> at SNP <span class="math notranslate nohighlight">\(j\)</span>.</p></li>
</ul>
<p>We will test for association between SNP <span class="math notranslate nohighlight">\(j\)</span> and the phenotype using the following linear regression model:</p>
<div class="math notranslate nohighlight">
\[
Y = \alpha_j + \beta_j X_j + \epsilon_j
\]</div>
<p>where:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(\alpha_j\)</span> is an intercept term. We will mostly ignore this.</p></li>
<li><p><span class="math notranslate nohighlight">\(\beta_j\)</span> is the effect of SNP <span class="math notranslate nohighlight">\(X_j\)</span>. It can be interpreted as the change in phenotype with each additional copy of the alternate allele.</p></li>
<li><p><span class="math notranslate nohighlight">\(\epsilon_j\)</span> is the noise term. Typically <span class="math notranslate nohighlight">\(\epsilon_j\)</span> is drawn from a multivariate normal distribution <span class="math notranslate nohighlight">\(\epsilon_j \sim N(0, \sigma_g^2 I)\)</span>.
That is, each individual’s <span class="math notranslate nohighlight">\(\epsilon_{ij}\)</span> is drawn from a normal distribution with mean 0 and variance <span class="math notranslate nohighlight">\(\sigma^2_g\)</span>. Because the off-diagonal elements of <span class="math notranslate nohighlight">\(\sigma^2_g I\)</span> are 0, we are not modeling any covariance between the error terms of different individuals (this will change when we talk about linear mixed models later).</p></li>
</ul>
<p>We can run a simple least squares linear regression to estimate the following key quantities for each SNP:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(\hat{\beta_j}\)</span>: the estimate of the effect size in the model above</p></li>
<li><p><span class="math notranslate nohighlight">\(p_j\)</span>: the p-value for SNP <span class="math notranslate nohighlight">\(j\)</span>, testing the null hypothesis that the SNP’s effect size is 0. We explore p-values in more detail in section 7.3.</p></li>
</ul>
<p>The quantities above (effect size, pvalue) are referred to as <em>summary statistics</em>. Unlike individual level data, which is often under access control due to privacy and other concerns, summary statistics in most cases can be freely published and shared.</p>
</section>
<section id="visualizing-gwas-results">
<h2>7.1.4 Visualizing GWAS results<a class="headerlink" href="#visualizing-gwas-results" title="Permalink to this heading">#</a></h2>
<p>Once summary statistics are obtained for each SNP, we can use several strategies to visualize them.</p>
<p>The plot below is a <em>Manhattan plot</em>. Each dot in the plot represents a single SNP that was tested. The x-axis gives the chromosome position. The y-axis gives the <span class="math notranslate nohighlight">\(-\log_{10}\)</span> p-value of the SNP. Note, we use <span class="math notranslate nohighlight">\(-\log_{10}\)</span> instead of the raw p-value since this emphasizes the most significant (lowest) p-values, which are the ones we are most interested in.</p>
<p><img alt="manhattan" src="../../_images/manhattan.png" />
Source: <a class="reference external" href="https://www.nature.com/articles/ncomms10448">Hu et al Nature Communications 2016</a> (“Are you a morning person”?)</p>
<p>We can notice an interesting pattern in the plot above: we often see “skyscrapers” of many strongly trait-associated SNPs nearby to each other, which inspired the name of Manhattan plot. This pattern is driven by the fact that SNPs that are nearby tend to be in <em>linkage disequilibrium (LD)</em>, and thus genotypes from nearby SNPs will be correlated with each other. If one SNP is causal for a trait, then it will be associated with the trait, but its LD partners will also be. An important conclusion that we can draw from this is that many trait-associated variants are unlikely to be causal themselves, but rather are just correlated with something else that is causal.</p>
<p>At the end of the day, each of these skyscrapers represents some underlying biological story for how genetic variation impacts a phenotype. When we discuss <em>fine-mapping</em>, we will explore the challenge of identifying the truly causal variants and extracting biological insights from these (it’s hard!). For now, we will focus on other insights we can get from the association statistics themselves, which we will eventually use to build models to predict phenotypes from genotypes.</p>
</section>
<section id="gwas-significance-threshold">
<h2>7.1.5 GWAS significance threshold<a class="headerlink" href="#gwas-significance-threshold" title="Permalink to this heading">#</a></h2>
<p>In GWAS, we are often testing millions of SNPs for association with a trait. Therefore we cannot use widely used p-value thresholds like <span class="math notranslate nohighlight">\(p&lt;0.01\)</span> or <span class="math notranslate nohighlight">\(p&lt;0.05\)</span> to determine which are significant. If we test 5 million SNPs, tens of thousands of them would pass a nominal significance threshold just by chance. We have to adjust for the fact that we performed many hypothesis tests to choose a more appropriate significance threshold.</p>
<p>A common method to control for multiple hypothesis testing is <em>Bonferroni correction</em>. The method is simple: if we performed <span class="math notranslate nohighlight">\(m\)</span> tests, and we want a significance level of <span class="math notranslate nohighlight">\(p&lt;0.01\)</span>, we should only reject the null hypothesis for tests at which their p-value was <span class="math notranslate nohighlight">\(&lt;0.01/m\)</span>.</p>
<p>However, adjusting by the total number of SNPs would be quite stringent. Since many of them are highly correlated (in strong LD), we would be over-correcting (imaging 5 SNPs which are in perfect LD. this represents effectively one test but we would count it 5 times).</p>
<p>It turns out there are around 1 million independent LD blocks in the human genome (this is why SNP genotyping arrays can genotype around only a million SNPs, since the rest can be imputed from those). So if we want a significance threshold of 0.05 Bonferroni-corrected for the effective number of independent tests, we end up with <span class="math notranslate nohighlight">\(0.05/1000000 = 5\times10^{-8}\)</span></p>
<p>This value, <span class="math notranslate nohighlight">\(p&lt;5\times10^{-8}\)</span>, is a widely used <em>genome-wide significance threshold</em> applied as standard practice in GWAS.</p>
</section>
<section id="more-on-gwas-qc">
<h2>7.1.6 More on GWAS QC<a class="headerlink" href="#more-on-gwas-qc" title="Permalink to this heading">#</a></h2>
<p>Although we will not discuss it in detail in this course, a critical component to performing GWAS on real data is to do careful quality control and filtering of the data to ensure the phenotypes and genotypes are high quality and reduce the chance of false positive/negative signals. Some important QC steps include:</p>
<ul class="simple">
<li><p><em>Sample-level QC</em>: e.g. removing related individuals, removing individuals with high rates of heterozygous genotypes indicating contamination, removing individuals with high rates of missing genotypes</p></li>
<li><p><em>Variant-level QC</em>: e.g. removing SNPs whose genotype do not follow expectation from Hardy-Weinberg Equilibrium, removing SNPs with high missingness rates or low minor allele frequencies</p></li>
<li><p><em>Call-level QC</em>: e.g. removing calls with low imputation or genotype quality scores.</p></li>
</ul>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./complextraits/gwas"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="../gwas.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Chapter 7: GWAS for complex traits</p>
      </div>
    </a>
    <a class="right-next"
       href="GWAS-simulations.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">7.2 Exploring GWAS association statistics</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#gwas-overview">7.1.2 GWAS Overview</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#association-testing-for-a-quantitative-trait">7.1.3 Association testing for a quantitative trait</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#visualizing-gwas-results">7.1.4 Visualizing GWAS results</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#gwas-significance-threshold">7.1.5 GWAS significance threshold</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#more-on-gwas-qc">7.1.6 More on GWAS QC</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Melissa Gymrek
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2022.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../_static/scripts/bootstrap.js?digest=8d27b9dea8ad943066ae"></script>
<script src="../../_static/scripts/pydata-sphinx-theme.js?digest=8d27b9dea8ad943066ae"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>