

<!DOCTYPE html>


<html lang="en" data-content_root="" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>4.2 Global ancestry analysis with ADMIXTURE &#8212; CSE284 - Personal Genomics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="../../_static/styles/bootstrap.css?digest=5b4479735964841361fd" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=5b4479735964841361fd" rel="stylesheet" />

  
  <link href="../../_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=5b4479735964841361fd" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css" />
    <link rel="stylesheet" href="../../_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=5b4479735964841361fd" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd" />
  <script src="../../_static/vendor/fontawesome/6.1.2/js/all.min.js?digest=5b4479735964841361fd"></script>

    <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/clipboard.min.js"></script>
    <script src="../../_static/copybutton.js"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../../_static/sphinx-thebe.js"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'ancestry/global/admixture_global_ancestry';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Chapter 5: Relative finding" href="../relative_finding.html" />
    <link rel="prev" title="2.3 Global ancestry analysis with PCA" href="pca_global_ancestry.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <div id="pst-scroll-pixel-helper"></div>

  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>
    Back to top
  </button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  

<a class="navbar-brand logo" href="../../index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../_static/logo.png" class="logo__image only-light" alt="CSE284 - Personal Genomics - Home"/>
    <script>document.write(`<img src="../../_static/logo.png" class="logo__image only-dark" alt="CSE284 - Personal Genomics - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../index.html">
                    Personal Genomics for Bioinformaticians
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 1 - Introduction</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/intro_to_your_genome.html">Chapter 1: Introduction to the human genome</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-1"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/structure_DNA.html">1.1 The structure of DNA</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/genome_organization.html">1.2 Organization of the human genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/central_dogma.html">1.3 Genes and the central dogma</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/noncoding.html">1.4 The non-coding genome</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/genome/review_questions.html">1.5 Review Questions</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/intro_genome_tech.html">Chapter 2: Overview of technologies for genome analysis</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-2"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/genotype_arrays.html">2.1 Genotyping arrays</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/ngs.html">2.2 Next-generation sequencing</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/long_reads.html">2.3 Long-read technologies</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/tech_summary.html">2.4 Summary of genomics technologies</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/tech/review_questions.html">2.5 Review Questions</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../intro/genome_file_formats.html">Tutorial 1: file formats for describing genetic variation</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-3"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/vcf.html">T1.1 VCF files</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/plink.html">T1.2 Plink files</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../intro/formats/pgen.html">T1.2 PGEN files</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 2 - Ancestry</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../intro_popgen.html">Chapter 3: Introduction to population genetics</a><input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-4"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../popgen/genetic_variation_terminology.html">3.1 Genetic variation terminology</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/forces_evolution.html">3.2 The four forces of evolution</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/mutations.html">3.3 Types and patterns of mutations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/drift.html">3.4 Genetic drift</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/selection.html">3.5 Selection</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/hwe.html">3.6 Hardy-Weinberg Equilibrium and random mating</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/fst.html">3.7 Measuring population differentiation with Fst</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/recombination.html">3.8 Recombination</a></li>
<li class="toctree-l2"><a class="reference internal" href="../popgen/review_questions.html">3.9 Review Questions</a></li>
</ul>
</li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="../global_ancestry.html">Chapter 4: Global ancestry</a><input checked="" class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-5"><i class="fa-solid fa-chevron-down"></i></label><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="pca_global_ancestry.html">2.3 Global ancestry analysis with PCA</a></li>


<li class="toctree-l2 current active"><a class="current reference internal" href="#">4.2 Global ancestry analysis with ADMIXTURE</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../relative_finding.html">Chapter 5: Relative finding</a><input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-6"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd.html">5.1 Identity by descent</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd_pedigree.html">5.2 Expected IBD of close relatives</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd_plink.html">5.3 Computing expected IBD - plink method</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/ibd_segments.html">5.4 Computing expected IBD segment sharing</a></li>
<li class="toctree-l2"><a class="reference internal" href="../relfind/genetic_genealogy.html">5.5 Genetic genealogy</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../admixture.html">Chapter 6: Admixture</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../plink_1000G.html">Tutorial 2: Working with 1000 Genomes data and plink</a><input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-7"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../plink1kg/1000G.html">T2.1 The 1000 Genomes Project dataset (and wrangling VCF files)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../plink1kg/plink.html">T2.2 Some tips and examples for using plink</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 3 - Complex traits</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../complextraits/overview.html">Chapter XX: Introduction to complex traits</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../complextraits/gwas.html">Chapter XX: Genome-wide association studies</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../complextraits/lmm.html">Chapter XX: Linear mixed models</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../complextraits/qtls.html">Chapter XX: Quantitative trait loci and functional genomics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../complextraits/fine-mapping.html">Chapter XX: Fine-mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../complextraits/prs.html">Chapter XX: Polygenic risk scores</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 4 - Trans-ancestry approaches</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../multianc/fine-mapping.html">Chapter XX: PRS and portability</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../multianc/prs.html">Chapter XX: PRS and portability</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 5 - Beyond common variants</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../beyond/rare.html">Chapter XX: Rare variants</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../beyond/complex.html">Chapter XX: Complex variants</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Module 6 - Ethics</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../ethics/privacy.html">Chapter XX: Ethics part I - privacy</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../ethics/health.html">Chapter XX: Ethics part II - health</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../ethics/equity.html">Chapter XX: Ethics part I - equity</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Appendix 1 - review of math+CS concepts</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../appendix/linear_algebra.html">Chapter A1.1 Review of linear algebra concepts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../appendix/hmm.html">Chapter A2.2: Hidden Markov Models</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Appendix 2 - glossary and notation</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../../appendix/unix.html">Chapter A2.1: Unix commands used</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../appendix/software.html">Chapter A2.2: List of software tools used</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../appendix/glossary.html">Chapter A2.3: Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../appendix/notation.html">Chapter A2.4: Notation used throughout the book</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/gymrek-lab/personal-genomics-textbook" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/gymrek-lab/personal-genomics-textbook/issues/new?title=Issue%20on%20page%20%2Fancestry/global/admixture_global_ancestry.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/ancestry/global/admixture_global_ancestry.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm navbar-btn theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch nav-link" data-mode="light"><i class="fa-solid fa-sun fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="dark"><i class="fa-solid fa-moon fa-lg"></i></span>
    <span class="theme-switch nav-link" data-mode="auto"><i class="fa-solid fa-circle-half-stroke fa-lg"></i></span>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>4.2 Global ancestry analysis with ADMIXTURE</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#the-admixture-model-overview">The ADMIXTURE model overview</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#writing-down-the-likelihood-of-q-and-f">Writing down the likelihood of <span class="math notranslate nohighlight">\(Q\)</span> and <span class="math notranslate nohighlight">\(F\)</span></a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#references">References</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="global-ancestry-analysis-with-admixture">
<h1>4.2 Global ancestry analysis with ADMIXTURE<a class="headerlink" href="#global-ancestry-analysis-with-admixture" title="Permalink to this heading">#</a></h1>
<p>In the previous section we used principal components analysis (PCA) to analyze global ancestry based on SNP genotypes. Here we explore an approach as implemented in the ADMIXTURE program [1] to characterize global ancestry in a different way.</p>
<p>The main idea of ADMIXTURE is to model each individual as coming from a mixture of different source populations. We can then quantify each person’s ancestry in terms of the relative contribution of each of the sources.</p>
<section id="the-admixture-model-overview">
<h2>The ADMIXTURE model overview<a class="headerlink" href="#the-admixture-model-overview" title="Permalink to this heading">#</a></h2>
<p>The major components of the ADMIXTURE model are:</p>
<ul class="simple">
<li><p>There are <span class="math notranslate nohighlight">\(K\)</span> “source populations”. We have to choose <span class="math notranslate nohighlight">\(K\)</span> up front. But, we don’t need to know what the <span class="math notranslate nohighlight">\(K\)</span> populations actually correspond to. i.e. we don’t need to say one population is European, one is African, etc. We will instead learn what those source populations are.</p></li>
<li><p><span class="math notranslate nohighlight">\(q_{ik}\)</span> gives the fraction of individual <span class="math notranslate nohighlight">\(i\)</span>’s genome that is from population <span class="math notranslate nohighlight">\(k\)</span>. For example, we might have individual <span class="math notranslate nohighlight">\(i\)</span> who is a mixture of 50% from population 1 and 50% from population 2. (<strong>unknown</strong>)</p></li>
<li><p><span class="math notranslate nohighlight">\(f_{kj}\)</span> gives the minor allele frequency of SNP <span class="math notranslate nohighlight">\(j\)</span> in population <span class="math notranslate nohighlight">\(k\)</span>. For example, we might have a SNP with MAF of 0% in population 1 and 10% in population 2. (<strong>unknown</strong>)</p></li>
</ul>
<p>In running ADMIXTURE, we will set <span class="math notranslate nohighlight">\(K\)</span> to a fixed value (e.g. 3). Then, we will attempt to infer <span class="math notranslate nohighlight">\(Q={q_ik}\)</span> and <span class="math notranslate nohighlight">\(F={f_{kj}}\)</span> from the data. Below we go through how to frame this as a maximum likelihood problem.</p>
</section>
<section id="writing-down-the-likelihood-of-q-and-f">
<h2>Writing down the likelihood of <span class="math notranslate nohighlight">\(Q\)</span> and <span class="math notranslate nohighlight">\(F\)</span><a class="headerlink" href="#writing-down-the-likelihood-of-q-and-f" title="Permalink to this heading">#</a></h2>
<p>Recall from our discussion of Hardy-Weinberg Equilibrium that if we know the MAF <span class="math notranslate nohighlight">\(p\)</span>, we can easily compute the frequency of the three possible genotypes at a bi-allelic locus:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(P(00) = (1-p)^2\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(P(01) = 2p(1-p)\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(P(11) = (1-p)^2\)</span></p></li>
</ul>
<p>Here, we extend the same logic to compute the probability of each possible genotype for each SNP, but taking into account that each individual is a mixture of different source populations. Instead of using <span class="math notranslate nohighlight">\(p\)</span>, we will use <span class="math notranslate nohighlight">\(\sum_{k}q_{ik}f_{kj}\)</span>, which computes a weighted MAF, weighted by the ancestry components of a particular individual. So for a single sample we get:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(P(00) = [\sum_{k}q_{ik}(1-f_{kj})]^2\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(P(01) = 2[\sum_{k}q_{ik}f_{kj}][\sum_{k}q_{ik}(1-f_{kj})]\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(P(11) = [\sum_{k}q_{ik}f_{kj}]^2\)</span></p></li>
</ul>
<p>Note, we can conveniently write this as a single equation. Let <span class="math notranslate nohighlight">\(g_{ij}\)</span> be the genotype (number of copies of the minor allele at SNP <span class="math notranslate nohighlight">\(j\)</span> in sample $i). Then:</p>
<div class="math notranslate nohighlight">
\[ P(g_{ij}|Q={q_{ij}}, F={f_{kj}}) = [\sum_{k}q_{ik}f_{kj}]^{g_{ij}}[\sum_{k}q_{ik}(1-f_{kj})]^{2-g_{ij}} \]</div>
<p>We can compute the likelihood of the entire data by assuming each SNP and each sample are independent. The goal is to find <span class="math notranslate nohighlight">\(Q\)</span> and <span class="math notranslate nohighlight">\(F\)</span> that maximize the likelihood below:</p>
<div class="math notranslate nohighlight">
\[L(Q, F) = \Pi_i \Pi_j P(g_{ij}|Q, F) \]</div>
<p>Note, when perform likelihood maximization, it is usually best to work with the log of the likelihood function. This is convenient, since it helps with numerical precision issues by avoiding really small numbers. It also turns products into sums which are easier to work with. Below we derive an equation for the log likelihood:</p>
<div class="math notranslate nohighlight">
\[\log L(Q, F) = \log \Pi_i \Pi_j P(g_{ij}|Q, F) \]</div>
<p>Using <span class="math notranslate nohighlight">\(\log AB = \log A + \log B\)</span> we get:</p>
<div class="math notranslate nohighlight">
\[\log L(Q, F) = \sum_i \log \Pi_j P(g_{ij}|Q, F) \]</div>
<p>and applying that rule again:</p>
<div class="math notranslate nohighlight">
\[\log L(Q, F) = \sum_i \sum_j \log P(g_{ij}|Q, F) \]</div>
<p>Note:</p>
<div class="math notranslate nohighlight">
\[ \log P(g_{ij}|Q, F) = \log [\sum_{k}q_{ik}f_{kj}]^{g_{ij}}[\sum_{k}q_{ik}(1-f_{kj})]^{2-g_{ij}} \]</div>
<p>Using <span class="math notranslate nohighlight">\(\log A^k = k\log A\)</span>: we get:</p>
<div class="math notranslate nohighlight">
\[ \log P(g_{ij}|Q, F) = g_{ij} \log [\sum_{k}q_{ik}f_{kj}] + (2-g_{ij})\log [\sum_{k}q_{ik}(1-f_{kj})] \]</div>
<p>Plugging this back into our main likelihood equation we get:</p>
<div class="math notranslate nohighlight">
\[L(Q, F) = \sum_i \sum_j  g_{ij} \log[\sum_{k}q_{ik}f_{kj}] + (2-g_{ij})\log [\sum_{k}q_{ik}(1-f_{kj})] \]</div>
<p>(which matches the likelihood equation given in Alexander, et al.)</p>
<p>We then must find values for <span class="math notranslate nohighlight">\(Q\)</span> and <span class="math notranslate nohighlight">\(F\)</span> to maximize this log likelihood. These are further subject to the constraints:</p>
<ul class="simple">
<li><p><span class="math notranslate nohighlight">\(q_{ik} \geq 0\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(\sum_k q_{ik} = 1\)</span></p></li>
<li><p><span class="math notranslate nohighlight">\(0 \leq f_{kj} \leq 1\)</span></p></li>
</ul>
<p>Note, this is a very high dimensional problem! There are often hundreds of samples and tens of thousands of SNPs in a single ADMIXTURE run. We do not go through the procedure to optimize this function, but that is a major innovation of the ADMIXTURE program [1].</p>
<p>In the problem set, you’ll explore using ADMIXTURE to characterize global ancestry in the 1000 Genomes Project samples. See [2] for an example.</p>
</section>
<section id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this heading">#</a></h2>
<p>[1] Fast model-based estimation of ancestry in unrelated individuals. Alexander et al. Genome Research 2009.</p>
<p>[2] <a class="reference external" href="https://www.nature.com/articles/nature15393/figures/2">https://www.nature.com/articles/nature15393/figures/2</a></p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./ancestry/global"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="pca_global_ancestry.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">2.3 Global ancestry analysis with PCA</p>
      </div>
    </a>
    <a class="right-next"
       href="../relative_finding.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Chapter 5: Relative finding</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#the-admixture-model-overview">The ADMIXTURE model overview</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#writing-down-the-likelihood-of-q-and-f">Writing down the likelihood of <span class="math notranslate nohighlight">\(Q\)</span> and <span class="math notranslate nohighlight">\(F\)</span></a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#references">References</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Melissa Gymrek
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2022.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../_static/scripts/bootstrap.js?digest=5b4479735964841361fd"></script>
<script src="../../_static/scripts/pydata-sphinx-theme.js?digest=5b4479735964841361fd"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>